packages/react-link/src/components/Link/Link.types.ts

```diff
@@ -1,73 +1,78 @@
 
-    /**
-     * Where to open the linked URL. Common values are `_blank` (a new tab or window),
-     * `_self` (the current window/context), `_parent`, and `_top`.
-     * This prop is only applied if `href` is set.
-     */
-    target?: string;
+  /**
+   * URL the link points to. If not provided, the link renders as a button (unless that behavior is
+   * overridden using `as`).
+   */
+  href?: string;
 
-    /**
-     * Built-in HTML attribute with different behavior depending on how the link is rendered.
-     * If rendered as `<a>`, hints at the MIME type.
-     * If rendered as `<button>`, override the type of button (`button` is the default).
-     */
-    type?: string;
+  /**
+   * If true, changes styling when the link is being used alongside other text content.
+   * @default false
+   */
+  inline?: boolean;
 
-    /**
-     * Whether the link is disabled.
-     * @default false
-     */
-    disabled?: boolean;
+  /**
+   * Click handler for the link.
+   */
+  onClick?: (ev: React.MouseEvent<HTMLAnchorElement> & React.MouseEvent<HTMLButtonElement>) => void;
 
-    /**
-     * When set, allows the link to be focusable even when it has been disabled. This is used in scenarios where it is
-     * important to keep a consistent tab order for screen reader and keyboard users.
-     * @default false
-     */
-    disabledFocusable?: boolean;
+  /**
+   * Relationship to the linked URL (can be a space-separated list).
+   * Most common values are `noreferrer` and/or `noopener`.
+   * This prop is only applied if link is rendered as `<a>`.
+   */
+  rel?: string;
 
-    /**
-     * If true, changes styling when the link is being used alongside other text content.
-     * @default false
-     */
-    inline?: boolean;
+  /**
+   * WAI-ARIA role applied to the element.
+   * This prop is only applied if link is rendered as `<button>`.
+   * @default "link"
+   */
+  role?: string;
 
-    /**
-     * If true, changes the link styling to emphasize that it represents an alternative action.
-     * @default false
-     */
-    secondary?: boolean;
-  };
+  /**
+   * If true, changes the link styling to emphasize that it represents an alternative action.
+   * @default false
+   */
+  secondary?: boolean;
 
-/**
- * {@docCategory Link}
- */
-export interface LinkState extends LinkProps {
-  ref: React.Ref<HTMLElement>;
-}
+  /**
+   * Where to open the linked URL. Common values are `_blank` (a new tab or window),
+   * `_self` (the current window/context), `_parent`, and `_top`.
+   * This prop is only applied if link is rendered as `<a>`.
+   */
+  target?: string;
+
+  /**
+   * Built-in HTML attribute with different behavior depending on how the link is rendered.
+   * If rendered as `<a>`, hints at the MIME type.
+   * If rendered as `<button>`, override the type of button (`button` is the default).
+   */
+  type?: string;
+};
+
+export type LinkProps = Omit<ComponentProps<LinkSlots>, 'type'> & LinkCommons & {};
```

Comment:
This has typically been an interface. I don't know if there's a strong reason to do it one way or the other, aside from consistency:
```suggestion
export interface LinkProps extends Omit<ComponentProps<LinkSlots>, 'type'>, LinkCommons {}
```

Reply:
The biggest difference between `types` and `interface` are that `interfaces` are open and `types` are closed for modifications. The other big one is that `interface` is not dynamically processed while `types` are, in other words: `types`  works best with `unions`

Reply:
Yes, it is inevitable in here.